api/middleware/rateLimiter/index.test.js

```diff
@@ -0,0 +1,126 @@
+    const originalEnv = process.env.NODE_ENV;
+    process.env.NODE_ENV = 'production';
+    try {
+        // Use a low max for testing
+        const limiter = rateLimit({
+            windowMs: 60 * 1000,
+            max: 5,
+            skip: () => false,
+            keyGenerator: (req, res) => {
+                const authHeader = req.headers.authorization;
+                if (authHeader && authHeader.startsWith('Bearer ')) {
+                    return authHeader.substring(7);
+                }
+                throw new Error('Missing JWT: rejecting request');
+            },
+            handler: (req, res) => {
+                res.status(429).send('Too many requests, please try again later');
+            }
+        });
+        const app = createTestApp(limiter);
+        const token = 'test-token';
+        let lastRes;
+        for (let i = 0; i < 10; i++) {
+            lastRes = await request(app).get('/api/test').set('Authorization', `Bearer ${token}`);
+        }
+        assert.strictEqual(lastRes.status, 429, 'Should return 429 when rate limit exceeded');
+        assert.match(lastRes.text, /Too many requests/i);
+    } finally {
+        process.env.NODE_ENV = originalEnv;
+    }
+});
+
+test('throws error if JWT is missing', async () => {
+    const originalEnv = process.env.NODE_ENV;
+    process.env.NODE_ENV = 'production';
+    try {
+        // Create a limiter that doesn't skip missing auth (for testing the rate limiter behavior)
+        const limiter = rateLimit({
+            windowMs: 60 * 1000,
+            max: 100,
+            skip: (req, res) => process.env.NODE_ENV !== 'production', // Check NODE_ENV directly
+            keyGenerator: (req, res) => {
+                const authHeader = req.headers.authorization;
+                if (authHeader && authHeader.startsWith('Bearer ')) {
+                    return authHeader.substring(7);
+                }
+                const err = new Error('Missing JWT: rejecting request');
+                err.status = 401;
+                throw err;
+            },
+            handler: (req, res) => {
+                throw new Error('Too many requests, please try again later');
+            }
+        });
+
+        const app = createTestApp(limiter);
+        const res = await request(app).get('/api/test');
+        assert.strictEqual(res.status, 401);
+        assert.match(res.text, /Missing JWT: rejecting request/i);
+    } finally {
+        process.env.NODE_ENV = originalEnv;
+    }
+});
+
+test('skips rate limiting in non-production environments', async () => {
+    const originalEnv = process.env.NODE_ENV;
+    process.env.NODE_ENV = 'development';
+    try {
+        const app = createTestApp(rateLimiter);
+        const res = await request(app).get('/api/test');
+        // Should skip and hit the route, return 200
+        assert.strictEqual(res.status, 200);
+    } finally {
+        process.env.NODE_ENV = originalEnv;
```

Comment:
Do the set up and tear down of env-related stuff outwith the test. In a before(Each) and after(Each).